irohad/subscription/dispatcher.hpp

```diff
@@ -12,90 +12,27 @@
 namespace iroha::subscription {
 
   template <uint32_t kCount, uint32_t kPoolSize>
-  class Dispatcher final : utils::NoCopy, utils::NoMove {
-   public:
-    static constexpr uint32_t kHandlersCount = kCount;
-    static constexpr uint32_t kPoolThreadsCount = kPoolSize;
-    using Task = ThreadHandler::Task;
+  struct IDispatcher {
```

Comment:
I'd suggest to make constexpr constructor with these parameters. And keep IDispatcher not a template.

Reply:
IDispatcher is a template to make compile-time check of a executor thread in subscription calls.